### PR TITLE
SONARJAVA-4413 Vert.x methods for JUnit5 in AssertionsInTestsCheck

### PR DESCRIPTION
**Pull request by julianladisch**
Add these methods of Eclipse Vert.x with JUnit 5 (VertxTestContext)
to list of assertion methods:
* laxCheckpoint
* succeedingThenComplete

Exclude these methods of Eclipse Vert.x with JUnit 5 (VertxTestContext)
from list of assertion methods:
* verify
* failing

Upgrade Vert.x version from 3.5.4 to 4.3.1. This requires code changes
in VertxJUnit4.java because HttpClient changed in Vert.x 4.

Use vertx-stack-depchain dependencyManagement, this also
manages netty-all, jackson-annotations, jackson-databind.

### CODE REVIEW

**Comment by johann-beleites-sonarsource** [comment]
Tracked in [SONARJAVA-4413](https://sonarsource.atlassian.net/browse/SONARJAVA-4413)

[SONARJAVA-4413]: https://sonarsource.atlassian.net/browse/SONARJAVA-4413?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by ADarko22** [comment]
@julianladisch, thanks for your PR; it looks nice! Could you please rebase it on `master` and resolve the conflicts? Most conflicts come from splitting the **java-checks-test-sources** module into submodules: you may need to add the required dependencies in the **java-checks-test-sources/default** submodule.

Once you solve those conflicts, I will merge your changes.

**Review by ADarko22 — CHANGES_REQUESTED** [review]
Please resolve the conflicts, then we will review the changes that overall look very nice!

File: java-checks/src/main/java/org/sonar/java/checks/helpers/UnitTestUtils.java [hunk 1]
```diff
@@ -39,8 +40,11 @@
 public final class UnitTestUtils {
 
   private static final String ORG_JUNIT_TEST = "org.junit.Test";
-  public static final Pattern ASSERTION_METHODS_PATTERN = Pattern.compile("(assert|verify|fail|should|check|expect|validate).*");
-  public static final Pattern TEST_METHODS_PATTERN = Pattern.compile("test.*|.*Test");
+  private static final Pattern ASSERTION_METHODS_PATTERN = Pattern.compile(
+      "(assert|verify|fail|should|check|expect|validate).*" +
+      "|laxCheckpoint|succeedingThenComplete" // Eclipse Vert.x with JUnit 5 (VertxTestContext)
```
Comment (ADarko22) [inline java-checks/src/main/java/org/sonar/java/checks/helpers/UnitTestUtils.java:45]: please move the comment on the line above

**Review by ADarko22 — CHANGES_REQUESTED** [review]

**Comment by ADarko22** [comment]
> @ADarko22 I've rebased the PR.

@julianladisch thank you! I left two small comments, please have a look then everything should be ready for merge :)

**Pull request closed without merging**